src/screens/SignIn/SignInScreen.js

```diff
@@ -1,10 +1,157 @@
-import React, { Component } from 'react';
-import { Text, View } from 'react-native';
+import React, { useState } from 'react';
+import {
+  StyleSheet, ActivityIndicator, View, Text, Alert,
+} from 'react-native';
+import { Button, Input, Icon } from 'react-native-elements';
+import auth from '@react-native-firebase/auth';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: 'center',
+    alignItems: 'center',
+  },
+  formContainer: {
+    height: 400,
+    padding: 20,
+  },
+  subContainer: {
+    marginBottom: 20,
+    padding: 5,
+  },
+  activity: {
+    position: 'absolute',
+    left: 0,
+    right: 0,
+    top: 0,
+    bottom: 0,
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  textInput: {
+    fontSize: 18,
+    margin: 5,
+    width: 200,
+  },
+});
 
 export default function SignInScreen({ navigation }) {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [showLoading, setShowLoading] = useState(false);
+  const signin = async () => {
+    setShowLoading(true);
+    try {
+      const doSignIn = await auth().signInWithEmailAndPassword(email, password);
+      setShowLoading(false);
+      if (doSignIn.user) {
+        navigation.navigate('Home');
+      }
+    } catch (e) {
+      setShowLoading(false);
+      Alert.alert(e.message);
+    }
+  };
+
   return (
-    <View>
-      <Text>Hello World! I hate navigation, I love react-native...</Text>
+    <View style={styles.container}>
+      <View style={styles.formContainer}>
+        <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
+          <Text style={{ fontSize: 28, height: 50 }}>Please Sign In!</Text>
+        </View>
+        <View style={styles.subContainer}>
+          <Input
+            style={styles.textInput}
+            placeholder="Your Email"
+            leftIcon={(
+              <Icon
+                name="mail"
+                size={24}
+              />
+                      )}
```

Comment:
there are some weird whitespace inconsistencies here, i think the ')}' should be on the same column as 'leftIcon'